stories/primer/details_component_stories.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class Primer::DetailsComponentStories < ViewComponent::Storybook::Stories
+  layout "storybook_centered_preview"
+
+  story(:details) do
+    controls do
+      select(:overlay, Primer::StoriesHelper.array_to_options(Primer::DetailsComponent::OVERLAY_MAPPINGS.keys), :none)
+      reset false
+    end
+
+    content do |component|
+      component.slot(:summary) { "Click me" }
+      component.slot(:body) { "Body" }
+    end
+  end
+
+  story(:custom_button) do
+    controls do
+      select(:overlay, Primer::StoriesHelper.array_to_options(Primer::DetailsComponent::OVERLAY_MAPPINGS.keys), :none)
+    end
+
+    content do |component|
+      component.slot(:summary, variant: :small, button_type: :primary) { "Click me" }
+      component.slot(:body) { "Body" }
+    end
+  end
+
+  story(:without_button) do
+    controls do
+      select(:overlay, Primer::StoriesHelper.array_to_options(Primer::DetailsComponent::OVERLAY_MAPPINGS.keys), :none)
+    end
+
+    content do |component|
+      component.slot(:summary, button: false) { "Click me" }
```

Comment:
What does it generate with `button: false`? It looks like because `kwargs[:tag]` is still `:summary`, `return Primer::BaseComponent.new(**kwargs) unless button` will still return a `<summary>` tag.

Reply:
it will still be a `summary` tag, the difference is the **component** being used.

In case we have `button: true` we are using a `Primer::ButtonComponent`, which has its own `variant`, `button_type` and other customizations

when using `button: false` it will use a `Primer::BaseComponent`, which allows a more basic implementation, without all the `Button` classes involved.

The separation was made to suppor the old `button: :reset` that was used

Reply:
Cool! So if I wanted just `<summary class="btn-link">`, I could do `button: false, classes: "btn-link"`?

Reply:
you could!